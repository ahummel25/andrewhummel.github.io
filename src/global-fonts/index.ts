import RobotoSansSerif from './KFOmCnqEu92Fr1Mu72xKKTU1Kvnz.woff2';

// injectGlobal`
// @font-face {
// 	font-family: 'Roboto-thin';
// 	font-style: normal;
// 	font-weight: 100;
// 	src: url('./KFOmCnqEu92Fr1Mu72xKKTU1Kvnz.woff2') format('woff2');
//   }
// `;

export { RobotoSansSerif };
