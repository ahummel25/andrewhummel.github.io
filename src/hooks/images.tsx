import { graphql, useStaticQuery } from "gatsby";

import { ImageProfileData } from "../interfaces";

export const useGeti11Image = (): ImageProfileData => {
  const imgData = useStaticQuery(
    graphql`
      query {
        file(relativePath: { eq: "i11.png" }) {
          childImageSharp {
            fluid(quality: 100, maxWidth: 60, maxHeight: 60) {
              ...GatsbyImageSharpFluid
            }
          }
        }
      }
    `
  );
  return imgData;
};
