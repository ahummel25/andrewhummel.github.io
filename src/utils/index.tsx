import { RefObject } from 'react';

export const handleScrollTo = (elRef: RefObject<HTMLDivElement>): void => {
  if (elRef.current) {
    //Scroll the element into view
    elRef.current.scrollIntoView({
      behavior: 'smooth',
      block: 'start'
    });
  }
};
