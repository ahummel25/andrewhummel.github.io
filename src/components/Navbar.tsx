import React, { FC, useEffect, useState } from 'react';
import styled from '@emotion/styled';
import Drawer from '@material-ui/core/Drawer';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Divider from '@material-ui/core/Divider';

import { useWindowDimensions } from '../hooks';
import { NavbarProps, NavListProps } from '../interfaces';
import { colors, fonts } from '../styles/variables';

import WeatherCard from './WeatherCard';

type StyledLiProps = {
  color: string;
};

const DRAWER_WIDTH = 190;
const MAX_WINDOW_WIDTH = 992;

const DrawerContainer = styled(({ ...rest }) => <Drawer {...rest} />)`
  flex-shrink: 0;

  ul {
    margin: 0;
  }

  .name-title {
    h1 {
      color: ${colors.lightGreen};
      font-family: ${fonts.robotoSansSerif};
      font-weight: 100;
      font-size: 30.45px;
      margin: 0 0 5px 0;
    }

    span {
      font-size: 14px;
    }

    background: ${colors.brandBackground};
    display: inline-block;
    min-height: 100px;
    text-align: left;
    padding: 30px 0 30px 30px;
    width: ${DRAWER_WIDTH}px;
  }
`;

const AppBarCustom = styled(({ ...rest }) => <AppBar {...rest} />)`
  display: block;
  text-align: center;
  font-size: 22px;

  .name-title > a {
    color: rgb(0, 128, 115);
    display: block;
    margin-bottom: -7px;
    padding-top: 10px;
  }

  span {
    font-size: 14px;
    display: block;
    padding-bottom: 3px;
  }

  a > i {
    font-size: 35px;
    display: block;
    float: left;
    padding-top: 15px;
    margin-right: -30px;
  }

  .icon-green {
    color: rgb(0, 128, 115);
  }

  .tool-bar {
    display: block;
    text-align: center;
  }
`;

const DrawerLi = styled.li<StyledLiProps>`
  cursor: pointer;
  padding: 0;
  font-weight: 300;
  display: flex;
  align-items: center;

  @keyframes hover-on {
    from {
      background-color: rgb(239, 340, 241);
    }
    to {
      background-color: rgb(250, 250, 250);
    }
  }

  &:hover {
    border-left: 3px solid ${(props): string => props.color};
    animation: hover-on 0.75s;
    animation-fill-mode: forwards;
    will-change: opacity, transform;
  }

  a {
    color: rgb(51, 51, 51);
    display: flex;
    align-items: center;
    padding: 10px 0 8px 10px;
    width: 100%;
  }

  i {
    color: rgb(51, 51, 51);
    font-weight: 500;
    display: flex;
    align-items: center;
    padding: 0 10px 0 5px;
    &.icon-color {
      color: ${(props): string => props.color};
    }
  }

  .active {
    background-color: rgb(250, 250, 250);
    border-left: 3px solid ${(props): string => props.color};
    font-weight: 500;
  }
`;

const navItems = [
  {
    text: 'About',
    icon: {
      type: 'person',
      color: 'rgb(0, 128, 115)'
    },
    ref: 'about'
  },
  {
    text: 'Experience',
    icon: {
      type: 'trending_up',
      color: 'rgb(0, 188, 212)'
    },
    ref: 'experience'
  },
  {
    text: 'Skills',
    icon: {
      type: 'equalizer',
      color: 'rgb(156,38,176)'
    },
    ref: 'skills'
  },
  {
    text: 'Certifications',
    icon: {
      type: 'star',
      color: 'rgb(244,67,53)'
    },
    ref: 'certifications'
  },
  {
    text: 'Contact',
    icon: {
      type: 'mail',
      color: 'rgb(121, 85, 72)'
    },
    ref: 'contact'
  }
];

const RenderFullDrawer: FC<Record<string, unknown>> = (): JSX.Element => (
  <DrawerContainer variant="permanent" anchor="left">
    <div className="name-title">
      <a href="/">
        <h1>
          Andrew <br /> Hummel
        </h1>
        <span className="brown-text light">Full Stack Developer</span>
      </a>
    </div>
    <Divider />
    <NavBarLists />
    <WeatherCard />
  </DrawerContainer>
);

const RenderMobileDrawer: FC<Record<string, unknown>> = (): JSX.Element => {
  const [open, setOpen] = useState<boolean>(false);

  const toggleDrawer = (
    event: React.KeyboardEvent | React.MouseEvent
  ): void => {
    event.preventDefault();

    if (
      event.type === 'keydown' &&
      ((event as React.KeyboardEvent).key === 'Tab' ||
        (event as React.KeyboardEvent).key === 'Shift')
    ) {
      return;
    }

    setOpen(!open);
  };

  return (
    <>
      <AppBarCustom color="inherit" position="fixed">
        <Toolbar className="tool-bar">
          <a href="/" onClick={toggleDrawer}>
            <i className="material-icons icon-green">menu</i>
          </a>
          <div className="name-title">
            <a href="/">Andrew Hummel</a>
            <span className="brown-text light">Full Stack Developer</span>
          </div>
        </Toolbar>
      </AppBarCustom>
      <DrawerContainer anchor="left" open={open} onClose={toggleDrawer}>
        <div className="name-title">
          <a href="/">
            <h1>Andrew Hummel</h1>
            <span className="brown-text light">Full Stack Developer</span>
          </a>
        </div>
        <Divider />
        <NavBarLists setOpen={setOpen} />
        <WeatherCard />
      </DrawerContainer>
    </>
  );
};

const NavBarLists: FC<NavListProps> = ({ setOpen }): JSX.Element => (
  <ul>
    {navItems.map(
      ({ text, icon: { type, color }, ref }, index): JSX.Element => (
        <DrawerLi key={index} color={color}>
          <a
            href={`#${ref}`}
            className="waves-effect waves-dark"
            onClick={(): void => {
              if (setOpen) setOpen(false);
            }}
          >
            <i className="small material-icons icon-color">{type}</i>
            {text}
          </a>
        </DrawerLi>
      )
    )}
  </ul>
);

const Navbar: FC<NavbarProps> = ({ setDrawerWidth }): JSX.Element => {
  const { width } = useWindowDimensions();
  const [showMobileNav, setShowMobileNav] = useState<boolean>(false);

  useEffect((): void => {
    setDrawerWidth(DRAWER_WIDTH);
  }, [setDrawerWidth]);

  useEffect((): void => {
    setShowMobileNav(width < MAX_WINDOW_WIDTH);
  }, [width]);

  return <>{showMobileNav ? <RenderMobileDrawer /> : <RenderFullDrawer />}</>;
};

export default Navbar;
