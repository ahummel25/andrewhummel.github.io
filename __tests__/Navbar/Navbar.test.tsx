jest.mock('../../src/hooks', () => ({
  useGetFixedFahrenheit: jest.fn(),
  useGetImages: jest.fn(),
  useWindowDimensions: jest.fn().mockImplementationOnce(() => ({
    width: 0,
    height: 0
  })),
  useGetWeatherByCoords: jest.fn()
}));

jest.mock('react-dom', () => {
  const reactDom = jest.requireActual('react-dom');
  return {
    ...reactDom,
    createPortal: (element: ReactNode): ReactNode => element
  };
});

// https://github.com/mui-org/material-ui/issues/12237
// jest.mock('@material-ui/core/Fade');
// jest.mock('@material-ui/core/Backdrop');
// jest.mock('@material-ui/core/Portal');
// jest.mock('@material-ui/core/Modal');
// jest.mock('@material-ui/core/Drawer');
// jest.mock('@material-ui/styles/withStyles');

jest.mock('@material-ui/core', () => {
  const materialUI = jest.requireActual('@material-ui/core');
  return {
    ...materialUI,
    Fade: jest.fn(({ children, open }) => (open ? children : null))
  };
});

// jest.mock('@material-ui/core/styles', () => ({
//   withStyles: (_styles: any) => (component: any) => component
// }));

import React, { ReactNode, ReactHTMLElement } from 'react';
import AppBar from '@material-ui/core/AppBar';
import Drawer from '@material-ui/core/Drawer';
import renderer from 'react-test-renderer';
import { mocked } from 'ts-jest/utils';

import Navbar from '../../src/components/Navbar';
import { useWindowDimensions } from '../../src/hooks';

const mockUseWindowDimensions = mocked(useWindowDimensions, true);

describe('Navbar', () => {
  const DRAWER_WIDTH = 190;
  const setDrawerWidth = jest.fn();

  it('renders correctly', () => {
    const tree = renderer.create(<Navbar setDrawerWidth={setDrawerWidth} />);

    expect(tree).toBeDefined();
  });

  describe('Mobile Navbar', () => {
    afterEach(() => {
      expect(mockUseWindowDimensions).toHaveBeenCalled();
      mockUseWindowDimensions.mockClear();
    });
    it('renders correctly with mobile version of drawer', () => {
      mockUseWindowDimensions.mockImplementation(() => ({
        width: 799,
        height: 799
      }));

      const tree = renderer.create(<Navbar setDrawerWidth={setDrawerWidth} />, {
        createNodeMock: (node: ReactHTMLElement<HTMLElement>): HTMLElement =>
          document.createElement(node.type)
      });

      expect(tree.root.props.setDrawerWidth).toHaveBeenCalledWith(DRAWER_WIDTH);
      expect(tree.root.props.setDrawerWidth.mock.calls[0][0]).toBe(
        DRAWER_WIDTH
      );

      const lis = tree.root.findAllByType('li');
      expect(lis.length).toBe(0);

      const mobileAppBar = tree.root.findAllByType(AppBar);
      expect(mobileAppBar.length).toBe(1);
      expect(mobileAppBar[0].props.color).toBe('inherit');
      expect(mobileAppBar[0].props.position).toBe('sticky');

      const mobileDrawer = tree.root.findAllByType(Drawer);
      expect(mobileDrawer.length).toBe(1);
      expect(mobileDrawer[0].props.onClose).toBeTruthy();
      expect(mobileDrawer[0].props.open).toBe(false);
      expect(mobileDrawer[0].props.children[2].props.setOpen).toBeTruthy();
      expect(typeof tree.root.props.setDrawerWidth).toBe('function');

      // @ts-ignore
      expect(tree.toJSON().children[0].children[0].props.onClick).toBeTruthy();

      //   const as = tree.root.findAllByType('a');
      //   const e = { preventDefault: jest.fn() };

      // Open mobile drawer
      /*act(() => {
        as[0].props.onClick(e);
      });

      expect(mobileDrawer[0].props.open).toBe(true);

      lis = tree.root.findAllByType('li');
      expect(lis.length).toBe(5);

      // Click an li and close mobile drawer
      act(() => {
        lis[0].props.children.props.onClick();
      });

      expect(mobileDrawer[0].props.open).toBe(false);*/
    });
  });

  describe('Desktop Navbar', () => {
    afterEach(() => {
      expect(mockUseWindowDimensions).toHaveBeenCalled();
      mockUseWindowDimensions.mockClear();
    });
    it('renders correctly with full version of drawer', () => {
      mockUseWindowDimensions.mockImplementation(() => ({
        width: 1200,
        height: 1200
      }));

      const tree = renderer.create(<Navbar setDrawerWidth={setDrawerWidth} />);

      expect(tree.root.props.setDrawerWidth).toHaveBeenCalledWith(DRAWER_WIDTH);
      expect(tree.root.props.setDrawerWidth.mock.calls[0][0]).toBe(
        DRAWER_WIDTH
      );

      const lis = tree.root.findAllByType('li');
      expect(lis.length).toBe(5);

      expect(tree.root.props.setDrawerWidth).toBe(setDrawerWidth);

      const treeJson = tree.toJSON();
      // @ts-ignore
      expect(treeJson.children[0].children[0].props).toStrictEqual({
        className: 'name-title'
      });
      // @ts-ignore
      expect(treeJson.children[0].children[0].props.onClick).toBeFalsy();
    });
  });
});
