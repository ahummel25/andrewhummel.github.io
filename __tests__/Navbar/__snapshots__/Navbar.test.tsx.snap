// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
.emotion-10 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-10 ul {
  margin: 0;
}

.emotion-10 .name-title {
  background: #fafafa;
  display: inline-block;
  min-height: 100px;
  text-align: left;
  padding: 30px 0 30px 30px;
  width: 190px;
}

.emotion-10 .name-title h1 {
  color: #008073;
  font-family: Roboto,sans-serif;
  font-weight: 100;
  font-size: 30.45px;
  margin: 0 0 5px 0;
}

.emotion-10 .name-title span {
  color: rgb(51,51,51);
  font-size: 14px;
}

.emotion-0 {
  cursor: pointer;
  padding: 0;
  font-weight: 300;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0:hover {
  border-left: 3px solid rgb(0,128,115);
  -webkit-animation: hover-on 0.75s;
  animation: hover-on 0.75s;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  will-change: opacity,transform;
}

.emotion-0 a {
  color: rgb(51,51,51);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px 0 8px 10px;
  width: 100%;
}

.emotion-0 i {
  color: rgb(51,51,51);
  font-weight: 500;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 10px 0 5px;
}

.emotion-0 i.icon-color {
  color: rgb(0,128,115);
}

.emotion-0 .active {
  background-color: rgb(250,250,250);
  border-left: 3px solid rgb(0,128,115);
  font-weight: 500;
}

.emotion-1 {
  cursor: pointer;
  padding: 0;
  font-weight: 300;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1:hover {
  border-left: 3px solid rgb(0,188,212);
  -webkit-animation: hover-on 0.75s;
  animation: hover-on 0.75s;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  will-change: opacity,transform;
}

.emotion-1 a {
  color: rgb(51,51,51);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px 0 8px 10px;
  width: 100%;
}

.emotion-1 i {
  color: rgb(51,51,51);
  font-weight: 500;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 10px 0 5px;
}

.emotion-1 i.icon-color {
  color: rgb(0,188,212);
}

.emotion-1 .active {
  background-color: rgb(250,250,250);
  border-left: 3px solid rgb(0,188,212);
  font-weight: 500;
}

.emotion-2 {
  cursor: pointer;
  padding: 0;
  font-weight: 300;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2:hover {
  border-left: 3px solid rgb(156,38,176);
  -webkit-animation: hover-on 0.75s;
  animation: hover-on 0.75s;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  will-change: opacity,transform;
}

.emotion-2 a {
  color: rgb(51,51,51);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px 0 8px 10px;
  width: 100%;
}

.emotion-2 i {
  color: rgb(51,51,51);
  font-weight: 500;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 10px 0 5px;
}

.emotion-2 i.icon-color {
  color: rgb(156,38,176);
}

.emotion-2 .active {
  background-color: rgb(250,250,250);
  border-left: 3px solid rgb(156,38,176);
  font-weight: 500;
}

.emotion-3 {
  cursor: pointer;
  padding: 0;
  font-weight: 300;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3:hover {
  border-left: 3px solid rgb(244,67,53);
  -webkit-animation: hover-on 0.75s;
  animation: hover-on 0.75s;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  will-change: opacity,transform;
}

.emotion-3 a {
  color: rgb(51,51,51);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px 0 8px 10px;
  width: 100%;
}

.emotion-3 i {
  color: rgb(51,51,51);
  font-weight: 500;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 10px 0 5px;
}

.emotion-3 i.icon-color {
  color: rgb(244,67,53);
}

.emotion-3 .active {
  background-color: rgb(250,250,250);
  border-left: 3px solid rgb(244,67,53);
  font-weight: 500;
}

.emotion-4 {
  cursor: pointer;
  padding: 0;
  font-weight: 300;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4:hover {
  border-left: 3px solid rgb(121,85,72);
  -webkit-animation: hover-on 0.75s;
  animation: hover-on 0.75s;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  will-change: opacity,transform;
}

.emotion-4 a {
  color: rgb(51,51,51);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px 0 8px 10px;
  width: 100%;
}

.emotion-4 i {
  color: rgb(51,51,51);
  font-weight: 500;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 10px 0 5px;
}

.emotion-4 i.icon-color {
  color: rgb(121,85,72);
}

.emotion-4 .active {
  background-color: rgb(250,250,250);
  border-left: 3px solid rgb(121,85,72);
  font-weight: 500;
}

.emotion-9 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  bottom: 10px;
  padding: 8px 0 0 8px;
  position: absolute;
  border: 1px solid rgba(0,0,0,0.3);
  border-radius: 12px;
  background-color: rgb(245,245,245);
  height: 150px;
  width: 90%;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 10px;
}

.emotion-5 {
  width: 50px;
  height: 50px;
  margin-right: 15px;
  border-radius: 50%;
  background-color: rgb(211,211,211);
}

@media (max-width:576px) {
  .emotion-5 {
    margin-right: 8px;
  }
}

.emotion-6 {
  font-size: 16px;
  font-weight: bold;
}

.emotion-8 {
  font-size: 16px;
  padding-bottom: 5px;
}

.emotion-10 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-10 ul {
  margin: 0;
}

.emotion-10 .name-title {
  background: #fafafa;
  display: inline-block;
  min-height: 100px;
  text-align: left;
  padding: 30px 0 30px 30px;
  width: 190px;
}

.emotion-10 .name-title h1 {
  color: #008073;
  font-family: Roboto,sans-serif;
  font-weight: 100;
  font-size: 30.45px;
  margin: 0 0 5px 0;
}

.emotion-10 .name-title span {
  color: rgb(51,51,51);
  font-size: 14px;
}

.emotion-0 {
  cursor: pointer;
  padding: 0;
  font-weight: 300;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0:hover {
  border-left: 3px solid rgb(0,128,115);
  -webkit-animation: hover-on 0.75s;
  animation: hover-on 0.75s;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  will-change: opacity,transform;
}

.emotion-0 a {
  color: rgb(51,51,51);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px 0 8px 10px;
  width: 100%;
}

.emotion-0 i {
  color: rgb(51,51,51);
  font-weight: 500;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 10px 0 5px;
}

.emotion-0 i.icon-color {
  color: rgb(0,128,115);
}

.emotion-0 .active {
  background-color: rgb(250,250,250);
  border-left: 3px solid rgb(0,128,115);
  font-weight: 500;
}

.emotion-1 {
  cursor: pointer;
  padding: 0;
  font-weight: 300;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1:hover {
  border-left: 3px solid rgb(0,188,212);
  -webkit-animation: hover-on 0.75s;
  animation: hover-on 0.75s;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  will-change: opacity,transform;
}

.emotion-1 a {
  color: rgb(51,51,51);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px 0 8px 10px;
  width: 100%;
}

.emotion-1 i {
  color: rgb(51,51,51);
  font-weight: 500;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 10px 0 5px;
}

.emotion-1 i.icon-color {
  color: rgb(0,188,212);
}

.emotion-1 .active {
  background-color: rgb(250,250,250);
  border-left: 3px solid rgb(0,188,212);
  font-weight: 500;
}

.emotion-2 {
  cursor: pointer;
  padding: 0;
  font-weight: 300;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2:hover {
  border-left: 3px solid rgb(156,38,176);
  -webkit-animation: hover-on 0.75s;
  animation: hover-on 0.75s;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  will-change: opacity,transform;
}

.emotion-2 a {
  color: rgb(51,51,51);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px 0 8px 10px;
  width: 100%;
}

.emotion-2 i {
  color: rgb(51,51,51);
  font-weight: 500;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 10px 0 5px;
}

.emotion-2 i.icon-color {
  color: rgb(156,38,176);
}

.emotion-2 .active {
  background-color: rgb(250,250,250);
  border-left: 3px solid rgb(156,38,176);
  font-weight: 500;
}

.emotion-3 {
  cursor: pointer;
  padding: 0;
  font-weight: 300;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3:hover {
  border-left: 3px solid rgb(244,67,53);
  -webkit-animation: hover-on 0.75s;
  animation: hover-on 0.75s;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  will-change: opacity,transform;
}

.emotion-3 a {
  color: rgb(51,51,51);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px 0 8px 10px;
  width: 100%;
}

.emotion-3 i {
  color: rgb(51,51,51);
  font-weight: 500;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 10px 0 5px;
}

.emotion-3 i.icon-color {
  color: rgb(244,67,53);
}

.emotion-3 .active {
  background-color: rgb(250,250,250);
  border-left: 3px solid rgb(244,67,53);
  font-weight: 500;
}

.emotion-4 {
  cursor: pointer;
  padding: 0;
  font-weight: 300;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4:hover {
  border-left: 3px solid rgb(121,85,72);
  -webkit-animation: hover-on 0.75s;
  animation: hover-on 0.75s;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  will-change: opacity,transform;
}

.emotion-4 a {
  color: rgb(51,51,51);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px 0 8px 10px;
  width: 100%;
}

.emotion-4 i {
  color: rgb(51,51,51);
  font-weight: 500;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 10px 0 5px;
}

.emotion-4 i.icon-color {
  color: rgb(121,85,72);
}

.emotion-4 .active {
  background-color: rgb(250,250,250);
  border-left: 3px solid rgb(121,85,72);
  font-weight: 500;
}

.emotion-9 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  bottom: 10px;
  padding: 8px 0 0 8px;
  position: absolute;
  border: 1px solid rgba(0,0,0,0.3);
  border-radius: 12px;
  background-color: rgb(245,245,245);
  height: 150px;
  width: 90%;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 10px;
}

.emotion-5 {
  width: 50px;
  height: 50px;
  margin-right: 15px;
  border-radius: 50%;
  background-color: rgb(211,211,211);
}

@media (max-width:576px) {
  .emotion-5 {
    margin-right: 8px;
  }
}

.emotion-6 {
  font-size: 16px;
  font-weight: bold;
}

.emotion-8 {
  font-size: 16px;
  padding-bottom: 5px;
}

<div
  className="MuiDrawer-root MuiDrawer-docked emotion-10"
>
  <div
    className="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
  >
    <div
      className="name-title"
    >
      <a
        href="/profile"
      >
        <h1>
          Andrew 
          <br />
           Hummel
        </h1>
        <span
          className="brown-text light"
        >
          Full Stack Developer
        </span>
      </a>
    </div>
    <hr
      className="MuiDivider-root"
    />
    <ul>
      <li
        className="emotion-0"
        color="rgb(0, 128, 115)"
      >
        <a
          className="waves-effect waves-dark"
          href="#about"
          onClick={[Function]}
        >
          <i
            className="small material-icons icon-color"
          >
            person
          </i>
          About
        </a>
      </li>
      <li
        className="emotion-1"
        color="rgb(0, 188, 212)"
      >
        <a
          className="waves-effect waves-dark"
          href="#experience"
          onClick={[Function]}
        >
          <i
            className="small material-icons icon-color"
          >
            trending_up
          </i>
          Experience
        </a>
      </li>
      <li
        className="emotion-2"
        color="rgb(156,38,176)"
      >
        <a
          className="waves-effect waves-dark"
          href="#skills"
          onClick={[Function]}
        >
          <i
            className="small material-icons icon-color"
          >
            equalizer
          </i>
          Skills
        </a>
      </li>
      <li
        className="emotion-3"
        color="rgb(244,67,53)"
      >
        <a
          className="waves-effect waves-dark"
          href="#certifications"
          onClick={[Function]}
        >
          <i
            className="small material-icons icon-color"
          >
            star
          </i>
          Certifications
        </a>
      </li>
      <li
        className="emotion-4"
        color="rgb(121, 85, 72)"
      >
        <a
          className="waves-effect waves-dark"
          href="#contact"
          onClick={[Function]}
        >
          <i
            className="small material-icons icon-color"
          >
            mail
          </i>
          Contact
        </a>
      </li>
    </ul>
    <div
      className="emotion-9"
    >
      <div
        className="emotion-7"
      >
        <img
          className="emotion-5"
          src=""
        />
        <div
          className="emotion-6"
        >
          Sun, Oct 4th
        </div>
      </div>
      <div
        className="emotion-8"
      >
        <strong>
           ℉
        </strong>
        | 
         ℉
      </div>
      <div>
        humidity: 
         %
      </div>
      <div>
        speed: 
         mps
      </div>
    </div>
  </div>
</div>
`;
